maps/mapList.lua

```diff
@@ -0,0 +1,20 @@
+--[[
+    This is a table of all maps available
+    All maps have a:
+    displayName - The display name of the map.
+    filename - The filename of the map that the game will load from.
+    thumbnail - A little picture to display when the map is selected.
+]]
```

Comment:
What if we just auto-load the maps from the maps/ directory like we do with entities? The displayName could be taken from the filename and a thumbnail with a matching filename could auto-load too. This would cut down on the amount of coupling between the source code and the map making (ie custom maps could essentially be drag n dropped). And any other properties you might think of later could even be stored under the map's "Custom Properties":

![Screenshot from 2022-01-24 13-29-32](https://user-images.githubusercontent.com/2344456/150842500-8585dde9-302f-4880-b87b-27b52c7e23e8.png)